extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CachedResultsProcessor.java

```diff
@@ -0,0 +1,255 @@
+            }
+
+            injectConfigs.add(new CachedResultsInjectConfig(cacheName, lockTimeout, keyGenerator, injectedClazz,
+                    annotation.target().declaredAnnotations(), excludes));
+        }
+
+        injectConfigs.stream().map(CachedResultsInjectConfigBuildItem::new).forEach(cachedResultsInjectConfigs::produce);
+    }
+
+    @BuildStep
+    void generateWrapperBeans(CombinedIndexBuildItem index,
+            List<CachedResultsInjectConfigBuildItem> cachedResultsInjectConfigs,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            BuildProducer<AdditionalCacheNameBuildItem> cacheNames) {
+        ClassOutput classOutput = new GeneratedBeanGizmo2Adaptor(generatedBeans);
+        Gizmo gizmo = Gizmo.create(classOutput);
+
+        for (CachedResultsInjectConfigBuildItem cachedResultInjectConfig : cachedResultsInjectConfigs) {
+            CachedResultsInjectConfig config = cachedResultInjectConfig.getConfig();
+            CompiledExcludes excludes = config.compileExcludes();
+            DotName clazzName = config.injectedClazz().name();
+            String name = clazzName.toString() + "_CachedResults_" + HashUtil.sha1(config.cacheName());
+            gizmo.class_(name, cc -> {
+                if (config.injectedClazz().isInterface()) {
+                    cc.implements_(classDescOf(clazzName));
+                } else {
+                    cc.extends_(classDescOf(clazzName));
+                }
+                cc.addAnnotation(Dependent.class);
+                cc.addAnnotation(CachedResults.class, ac -> {
+                    ac.add("cacheName", config.cacheName());
+                    if (!config.cacheName().equals(CachedResults.DEFAULT)) {
+                        cacheNames.produce(new AdditionalCacheNameBuildItem(config.cacheName()));
+                    }
+                });
+                cc.defaultConstructor();
+
+                FieldDesc delegateField = cc.field("cachedResults_delegate", fc -> {
+                    fc.packagePrivate();
+                    fc.setType(classDescOf(config.injectedClazz()));
+                    boolean hasInject = false;
+                    for (AnnotationInstance a : config.annotations()) {
+                        // copy all but @CachedResults
+                        if (a.name().equals(CACHED_RESULTS)) {
+                            continue;
+                        }
+                        if (a.name().equals(INJECT)) {
+                            hasInject = true;
+                        }
+                        addAnnotation(fc, a, index.getIndex());
+                    }
+                    if (!hasInject) {
+                        // Make sure cachedResults_delegate is an injection point
+                        fc.addAnnotation(Inject.class);
+                    }
+                });
+
+                for (MethodInfo method : config.injectedClazz().methods()) {
+                    if (method.isConstructor()
```

Comment:
I'm fairly sure I made a comment here, but I cannot find it anymore, so I'll repeat: you should also ignore `private` methods here.